### DISCUSSION TITLE
Importance of normalisation method for Reference Mapping

### DISCUSSION THREAD

**Discussion by chris-rands** [satijalab/seurat#4627] [Q&A]
I'm testing the nice reference mapping feature. It says "The reference was normalized using SCTransform(), so we use the same approach to normalize the query here." My question: is that important? I have a scanpy AnnData object that was processed with a scandard scanpy workflow (normalize_total, log1p, no regression/scaling), which I convert to h5seurat format- is it okay to use this to compare to the Seurat PBMC reference, which was normalised differently? The scanpy object does not hold the raw counts so I cannot re-normalize. The results look promising, thanks

(Cross-posted with issue https://github.com/satijalab/seurat/issues/4625, feel free to delete one of these threads)

**Comment by rsatija** [comment]
While you may get reasonable results when using log-normalization for query and SCTransform for reference, we don't recommend this.

The following script can be used to convert a log_normalized UMI matrix into a counts matrix. It only works if the original dataset was a UMI matrix, and therefore the smallest non-zero value in each cell vector represents 1 UMI.

You should be able to use this to convert the data in your AnnData object to a counts matrix, and then can map as described in the vignette.

```
# log-normalized matrix (i.e. data slot) is stored in lognorm_data

normdata = expm1(lognorm_data)
counts_matrix <- apply(normdata,MARGIN = 2,FUN = function(x) x=x/sort(x[x>0])[1])
```

**Reply by chris-rands** [reply]
Thank you, I didn't expect rsatija himself to pick-up my novice question! This produces some non-integer counts (e.g. 2.99999993208263), but perhaps only due to floating imprecision? I see `expm1` is the inverse of `log1p`, but need to dig deeper to understand how your last code line reverses the normalisation
